src/Stratis.Bitcoin.Features.Consensus.Tests/Rules/BlockHeaderRuleTest.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Threading.Tasks;
+using NBitcoin;
+using Stratis.Bitcoin.Features.Consensus.Rules.CommonRules;
+using Xunit;
+
+namespace Stratis.Bitcoin.Features.Consensus.Tests.Rules
+{
+    public class BlockHeaderRuleTest
+    {
+        [Fact]
+        public async Task BlockReceived_IsNextBlock_ValidationSucessAsync()
+        {
+            var testContext = TestRulesContextFactory.CreateAsync(Network.RegTest);
+            BlockHeaderRule blockHeaderRule = testContext.CreateRule<BlockHeaderRule>();
+
+            var context = new RuleContext(new BlockValidationContext(), Network.RegTest.Consensus, testContext.Chain.Tip);
+            context.BlockValidationContext.Block = new Block(new BlockHeader { HashPrevBlock = testContext.Chain.Tip.HashBlock });
+            await blockHeaderRule.RunAsync(context);
+
+            Assert.NotNull(context.BlockValidationContext.ChainedBlock);
+            Assert.NotNull(context.BestBlock);
+            Assert.NotNull(context.Flags);
+        }
+
+        [Fact]
+        public async Task BlockReceived_NotNextBlock_ValidationFailAsync()
+        {
+            var testContext = TestRulesContextFactory.CreateAsync(Network.RegTest);
+            BlockHeaderRule blockHeaderRule = testContext.CreateRule<BlockHeaderRule>();
+
+            var context = new RuleContext(new BlockValidationContext(), Network.RegTest.Consensus, testContext.Chain.Tip);
+            context.BlockValidationContext.Block = new Block(new BlockHeader { HashPrevBlock = uint256.Zero });
+            var error = await Assert.ThrowsAsync<ConsensusErrorException>(async () => await blockHeaderRule.RunAsync(context));
+
+            Assert.True(error.ConsensusError == ConsensusErrors.InvalidPrevTip);
```

Comment:
Think you might get a nicer error message if you do Assert.Equal(ConsensusErrors.InvalidPrevTip, error.ConsensusError)

Reply:
Ah yes thanks